shared/chat/conversation/messages/coinflip/results.js

```diff
@@ -0,0 +1,346 @@
+  {suit: 'diamonds', value: '6'},
+  {suit: 'diamonds', value: '7'},
+  {suit: 'diamonds', value: '8'},
+  {suit: 'diamonds', value: '9'},
+  {suit: 'diamonds', value: '10'},
+  {suit: 'diamonds', value: 'J'},
+  {suit: 'diamonds', value: 'Q'},
+  {suit: 'diamonds', value: 'K'},
+  {suit: 'diamonds', value: 'A'},
+  {suit: 'hearts', value: '2'},
+  {suit: 'hearts', value: '3'},
+  {suit: 'hearts', value: '4'},
+  {suit: 'hearts', value: '5'},
+  {suit: 'hearts', value: '6'},
+  {suit: 'hearts', value: '7'},
+  {suit: 'hearts', value: '8'},
+  {suit: 'hearts', value: '9'},
+  {suit: 'hearts', value: '10'},
+  {suit: 'hearts', value: 'J'},
+  {suit: 'hearts', value: 'Q'},
+  {suit: 'hearts', value: 'K'},
+  {suit: 'hearts', value: 'A'},
+]
+
+const suits = {
+  clubs: {
+    color: Styles.globalColors.black,
+    icon: 'iconfont-club',
+  },
+  diamonds: {
+    color: Styles.globalColors.red,
+    icon: 'iconfont-diamond',
+  },
+  hearts: {
+    color: Styles.globalColors.red,
+    icon: 'iconfont-heart',
+  },
+  spades: {
+    color: Styles.globalColors.black,
+    icon: 'iconfont-spade',
+  },
+}
+
+type CardType = {|
+  card: number,
+  hand?: boolean,
+|}
+const Card = (props: CardType) => (
+  <Kb.Box2
+    direction="vertical"
+    centerChildren={true}
+    gap={Styles.isMobile ? undefined : 'xtiny'}
+    style={Styles.collapseStyles([styles.card, !props.hand && styles.cardStacked])}
+  >
+    <Kb.Box2 direction="horizontal">
+      <Kb.Text
+        selectable={true}
+        type={Styles.isMobile ? 'BodyBig' : 'Header'}
+        style={{color: suits[cards[props.card].suit].color}}
+      >
+        {cards[props.card].value}
+      </Kb.Text>
+    </Kb.Box2>
+    <Kb.Box2 direction="horizontal">
+      <Kb.Icon
+        fontSize={16}
+        type={suits[cards[props.card].suit].icon}
+        color={suits[cards[props.card].suit].color}
+      />
+    </Kb.Box2>
+  </Kb.Box2>
+)
+
+type DeckType = {|
+  deck: ?Array<number>,
+  hand?: boolean,
+|}
+const CoinFlipResultDeck = (props: DeckType) => (
+  <Kb.Box2
+    direction="horizontal"
+    fullWidth={true}
+    style={Styles.collapseStyles([styles.cards, !props.hand && styles.cardsStacked])}
+  >
+    {props.deck && props.deck.map(card => <Card key={card} card={card} hand={props.hand} />)}
+  </Kb.Box2>
+)
+
+type CoinType = {|
+  coin: ?boolean,
+|}
+const CoinFlipResultCoin = (props: CoinType) => (
+  <Kb.Box2 direction="horizontal" fullWidth={true} gap="tiny">
+    <Kb.Box2 direction="vertical" style={styles.coin} centerChildren={true}>
+      <Kb.Icon
+        color={Styles.globalColors.white}
+        fontSize={props.coin ? 28 : 22}
+        style={styles.coinIcon}
+        type={props.coin ? 'iconfont-heads' : 'iconfont-tails'}
+      />
+    </Kb.Box2>
+    <Kb.Box2 direction="vertical" centerChildren={true}>
+      <Kb.Text selectable={true} type="Header">
+        {props.coin ? 'Heads!' : 'Tails!'}
+      </Kb.Text>
+    </Kb.Box2>
+  </Kb.Box2>
+)
+
+type HandType = {|
```

Comment:
just fyi you don't need any of these types inside this file. they'll all get inferred